nixos/modules/services/mail/postsrsd.nix

```diff
@@ -93,66 +230,120 @@ in
         default = "postsrsd";
         description = "Group for the daemon";
       };
-
-      socketPath = lib.mkOption {
-        type = lib.types.path;
-        default = "${runtimeDirectory}/socket";
-        readOnly = true;
-        description = ''
-          Path to the Unix socket for connecting to postsrsd.
-          Read-only, intended for usage when integrating postsrsd into other NixOS config.'';
-      };
     };
   };
 
-  config = lib.mkIf cfg.enable {
-    users.users = lib.optionalAttrs (cfg.user == "postsrsd") {
-      postsrsd = {
-        group = cfg.group;
-        uid = config.ids.uids.postsrsd;
+  config = lib.mkMerge [
+    (lib.mkIf (cfg.enable && cfg.configurePostfix && config.services.postfix.enable) {
+      services.postfix.config = {
+        # https://github.com/roehling/postsrsd#configuration
+        sender_canonical_maps = "socketmap:${cfg.settings.socketmap}:forward";
+        sender_canonical_classes = "envelope_sender";
+        recipient_canonical_maps = "socketmap:${cfg.settings.socketmap}:reverse";
+        recipient_canonical_classes = [
```

Comment:
This is probably related to the postfix module.

```
       error: The option `services.postfix.config.recipient_canonical_classes' is defined multiple times while it's expected to be unique.

       Definition values:
       - In `/nix/store/qd6rcfggxm1z6lqrj9riz9zd0q500d15-source/hosts/mail': "envelope_recipient, header_recipient"
       - In `/nix/store/p2fs4zm38yji76hk93birlyw97b7ikik-source/nixos/modules/services/mail/postsrsd.nix':
           [
             "envelope_recipient"
             "header_recipient"
           ]
       Use `lib.mkForce value` or `lib.mkDefault value` to change the priority on any of these definitions.
```

Reply:
That seems to be really unmaintained. Last commit is 9 years ago. Should we drop it in another PR? https://github.com/Fruneau/pfixtools